components/public-api-server/pkg/oidc/service.go

```diff
@@ -57,18 +57,20 @@ type AuthFlowResult struct {
 	Claims  map[string]interface{} `json:"claims"`
 }
 
-func NewService(sessionServiceAddress string, dbConn *gorm.DB, cipher db.Cipher) *Service {
+func NewService(sessionServiceAddress string, dbConn *gorm.DB, cipher db.Cipher, stateJWT *StateJWT) *Service {
 	return &Service{
 		verifierByIssuer:      map[string]*goidc.IDTokenVerifier{},
 		sessionServiceAddress: sessionServiceAddress,
 
 		dbConn: dbConn,
 		cipher: cipher,
+
+		stateJWT: stateJWT,
 	}
 }
 
-func newTestService(sessionServiceAddress string, dbConn *gorm.DB, cipher db.Cipher) *Service {
-	service := NewService(sessionServiceAddress, dbConn, cipher)
+func newTestService(sessionServiceAddress string, dbConn *gorm.DB, cipher db.Cipher, stateJWT *StateJWT) *Service {
+	service := NewService(sessionServiceAddress, dbConn, cipher, stateJWT)
 	service.skipVerifyIdToken = true
 	return service
 }
```

Comment:
Either move this to a test package, or make it a first class constructor which doens't mention Test. It's an anti-pattern in Go to have test code alongisde the non-test implementation.

But I think we shouldn't have the `skipVerifyIdToken` as a thing on this service at all. We should either abstract the verification behind an interface, such that we can mock/stub it in tests, or actually run the verification check.

Reply:
Alternatively, if you want to provide this as an option for the caller, encode it into the request such that the caller can choose to have the verification skipped.

Reply:
Let me know if you'd like to pair on this.

Reply:
'Removed `newTestService`.

`skipVerifyIdToken` is passed to a constructor from a library. It was a recommended way to use this shortcut and remove `skipVerifyIdToken` in a separate PR.

With the added `golang-jwt/jwt`, it's now easier to extend the mocked service to actually to the checks instead of skipping verification.

Reply:
> Let's find a better way to avoid the skipVerification, as that pushes test code into our main implementation.

`skipVerification` is deemed to be removed in following PR.